src/test/java/repository/CourseMongoRepositoryTest.java

```
@@ -0,0 +1,184 @@
+	}
+	
+	@Test
+	public void testFindCourseByIdShouldBeFound() {
+		//setup
+		addTestCourseToDatabase("id", "testCourse", Collections.emptyList());
+
+		//verify
+		assertThat(courseMongoRepository.findById("id"))
+		.isEqualTo(new Course("id", "testCourse"));
+	}
+	
+	@Test
+	public void testSave() {
+		//setup
+		Course testCourse = new Course("id", "testCourse");
+
+		//exercise
+		courseMongoRepository.save(testCourse);
+
+		//verify
+		assertThat(readAllCoursesFromDatabase()).containsExactly(testCourse);
+	}
+
+	@Test
+	public void testDelete() {
+		//setup
+		Course testCourse = new Course("id", "testCourse");
+
+		//exercise
+		courseMongoRepository.delete(testCourse);
+
+		//verify
+		assertThat(readAllCoursesFromDatabase()).isEmpty();
+	}
+	
+	@Test
+	public void testGetStudentsFromCourseWhenStudentListIsEmpty() {
+		//setup
+		addTestCourseToDatabase("idCourse", "testCourse", Collections.emptyList());
+
+		//exercise
+		List<String> courseStudents = courseMongoRepository.findCourseStudents("idCourse");
+
+		//verify
+		assertThat(courseStudents).isEqualTo(Collections.emptyList());
+	}
+	
+	@Test
+	public void testGetStudentsFromCourseWhenStudentListIsNotEmpty() {
+		//setup
+		addTestCourseToDatabase("idCourse", "testCourse", Collections.singletonList("idStudent"));
+
+		//exercise
+		List<String> courseStudents = courseMongoRepository.findCourseStudents("idCourse");
+
+		//verify
+		assertThat(courseStudents).isEqualTo(Collections.singletonList("idStudent"));
+	}
+	
+	@Test
+	public void testAddStudentToCourseWhenStudentIsNotNull() {
+		//setup
+		addTestCourseToDatabase("idCourse", "testCourse", Collections.emptyList());
+
+		//exercise
+		courseMongoRepository.updateCourseStudents("idStudent", "idCourse");
+		List<String> courseStudents = courseMongoRepository.findCourseStudents("idCourse");
+
+		//verify
+		assertThat(courseStudents).containsExactly("idStudent");
+	}
+
+	@Test
+	public void testRemoveCourseToStudentWhenCourseIsNotNull() {
```

Typo: `testRemoveCourseStudent...`
What about `testRemoveCourseStudentWhenCourseIsNull`?

---

Yes, I forgot to test that case. The removeCourseStudent should go in just fine and the findCourseStudents should return an emptyList()? Or we could throw an Exception..

---

Maybe throw an Exception is the best solution here. I don't know if this change will break some previous tests.